lib/adhearsion-ivr/ivr_controller.rb

```diff
@@ -22,6 +22,15 @@ def barge(val = nil)
         end
       end
 
+      # allow passing a grammar mode to MRCP for FS compatibility
+      def input_mode(val = nil)
+        if val.nil?
+          @input_mode || :voice
+        else
+          @input_mode = val
+        end
```

Comment:
This whole block could be simplified to `@input_mode = val || @input_mode || :voice`, though I would note that simply calling the method with no arguments (looking like an accessor) has the side effect of actually changing the instance's internal state by setting `@input_mode = :voice`